odoo/models.py

```diff
@@ -3075,13 +3077,14 @@ def qualify(field):
             fetched = self.browse(ids)
 
             for field in fields_pre:
-                values = next(cols)
+                values = list(next(cols))
                 if context.get('lang') and not field.inherited and callable(field.translate):
                     translate = field.get_trans_func(fetched)
-                    values = list(values)
                     for index in range(len(ids)):
                         values[index] = translate(ids[index], values[index])
 
+                for index in range(len(ids)):
+                    values[index] = field.convert_to_cache(values[index], fetched[index])
```

Comment:
Not calling `convert_to_cache` was an optimization.
Introducing it is therefore a counter-optimization!

The patched values should not be in cache.
You should override `convert_to_record` instead!

Reply:
Yes, I'm back to dev; testing and exploring.
But is that consistent? The cache can sometimes be filled with values that have been converted with the method and sometimes not.
I guess psycopg2 already does the conversions needed?

Reply:
We actually changed the types of values in cache to match the ones returned by `psycopg2`.
That is why the conversion is useless when values come from `cr.fetchall()`.